src/services/index.ts

```diff
@@ -0,0 +1,215 @@
+    storage_price: 'Balance',
+    file_map: 'Vec<(Vec<u8>, Vec<Hash>)>',
+  },
+  MerchantPunishment: {
+    success: 'EraIndex',
+    failed: 'EraIndex',
+    value: 'Balance',
+  },
+  MerkleRoot: 'Vec<u8>',
+  OrderStatus: {
+    _enum: ['Success', 'Failed', 'Pending'],
+  },
+  PaymentLedger: {
+    total: 'Balance',
+    paid: 'Balance',
+    unreserved: 'Balance',
+  },
+  Pledge: {
+    total: 'Balance',
+    used: 'Balance',
+  },
+  ReportSlot: 'u64',
+  Releases: {
+    _enum: ['V1_0_0', 'V2_0_0'],
+  },
+  SorderInfo: {
+    file_identifier: 'MerkleRoot',
+    file_size: 'u64',
+    created_on: 'BlockNumber',
+    merchant: 'AccountId',
+    client: 'AccountId',
+    amount: 'Balance',
+    duration: 'BlockNumber',
+  },
+  SorderStatus: {
+    completed_on: 'BlockNumber',
+    expired_on: 'BlockNumber',
+    status: 'OrderStatus',
+    claimed_at: 'BlockNumber',
+  },
+  SorderPunishment: {
+    success: 'BlockNumber',
+    failed: 'BlockNumber',
+    updated_at: 'BlockNumber',
+  },
+  Status: {
+    _enum: ['Free', 'Reserved'],
+  },
+  StorageOrder: {
+    file_identifier: 'Vec<u8>',
+    file_size: 'u64',
+    created_on: 'BlockNumber',
+    completed_on: 'BlockNumber',
+    expired_on: 'BlockNumber',
+    provider: 'AccountId',
+    client: 'AccountId',
+    amount: 'Balance',
+    order_status: 'OrderStatus',
+  },
+  SworkerCert: 'Vec<u8>',
+  SworkerCode: 'Vec<u8>',
+  SworkerPubKey: 'Vec<u8>',
+  SworkerSignature: 'Vec<u8>',
+  WorkReport: {
+    report_slot: 'u64',
+    used: 'u64',
+    free: 'u64',
+    files: 'BTreeMap<MerkleRoot, u64>',
+    reported_files_size: 'u64',
+    reported_srd_root: 'MerkleRoot',
+    reported_files_root: 'MerkleRoot',
+  },
+};
+
+// TODO: Better result
+export interface TxRes {
+  status?: string;
+  message?: string;
+  details?: string;
+}
+
+export const api = new ApiPromise({
+  provider: new WsProvider(process.argv[3] || 'ws://192.168.50.140:9931'),
+  types,
+});
+
+export const chain = {
+  header: (_: Request, res: Response, next: NextFunction) => {
+    withApiReady(async (api: ApiPromise) => {
+      const h = await header(api);
+      res.json({
+        number: h.number,
+        hash: h.hash,
+      });
+    }, next);
+  },
+  blockHash: (req: Request, res: Response, next: NextFunction) => {
+    withApiReady(async (api: ApiPromise) => {
+      res.send(await blockHash(api, Number(req.query['blockNumber'])));
+    }, next);
+  },
+  health: (_: Request, res: Response, next: NextFunction) => {
+    withApiReady(async (api: ApiPromise) => {
+      res.json(await health(api));
+    }, next);
+  },
+};
+
+export const swork = {
+  register: (req: Request, res: Response, next: NextFunction) => {
+    withApiReady(async (api: ApiPromise) => {
+      const krp = loadKeyringPair(req);
+      res.json(await register(api, krp, req));
+    }, next);
+  },
+  reportWorks: (req: Request, res: Response, next: NextFunction) => {
+    withApiReady(async (api: ApiPromise) => {
```

Comment:
withApiReady can be refactored into a middleware.
``` javascript
const apiMiddleware = (req, res, next) => {
  withApiReady((api) => {
   req.polkadotApi = api;
  next()
  })
}

app.use('/sworker', apiMiddleware)

```
then you can use req.polkadotApi in the handler

Reply:
Change it later.